File: backend/app/models/vehicle.py
from datetime import datetime


class Vehicle:
    """Domain model — not persisted directly; used for type clarity."""

    def __init__(
        self,
        id: str,
        make: str,
        model: str,
        category: str,
        price: float,
        quantity: int,
        created_at: datetime,
        updated_at: datetime,
    ) -> None:
        self.id = id
        self.make = make
        self.model = model
        self.category = category
        self.price = price
        self.quantity = quantity
        self.created_at = created_at
        self.updated_at = updated_at

File: frontend/src/types/vehicle.ts
export interface Vehicle {
  id: string;
  make: string;
  model: string;
  category: string;
  price: number;
  quantity: number;
  created_at: string;
  updated_at: string;
}

export interface VehicleListResponse {
  items: Vehicle[];
  total: number;
  page: number;
  limit: number;
}

export interface VehicleFilters {
  make?: string;
  model?: string;
  category?: string;
  min_price?: number;
  max_price?: number;
}

export interface VehiclePayload {
  make: string;
  model: string;
  category: string;
  price: number;
  quantity: number;
}

export type VehicleUpdatePayload = Partial<VehiclePayload>;

export type StockStatus = "In Stock" | "Low Stock" | "Out of Stock";

export const stockStatus = (quantity: number): StockStatus => {
  if (quantity === 0) return "Out of Stock";
  if (quantity <= 3) return "Low Stock";
  return "In Stock";
};
